community/consistency-check/src/main/java/org/neo4j/consistency/checking/SchemaRecordCheck.java

```diff
@@ -256,11 +258,22 @@ public void checkRelationshipPropertyExistenceRule( RelationshipPropertyExistenc
         }
     }
 
-    private void checkLabelAndPropertyRule( SchemaRule rule, int propertyKey, DynamicRecord record,
+    private void checkLabelAndPropertyRule( SchemaRule rule, EntityPropertyDescriptor descriptor, DynamicRecord record,
             RecordAccess records, CheckerEngine<DynamicRecord,ConsistencyReport.SchemaConsistencyReport> engine )
     {
         engine.comparativeCheck( records.label( rule.getLabel() ), VALID_LABEL );
-        engine.comparativeCheck( records.propertyKey( propertyKey ), VALID_PROPERTY_KEY );
+        if ( descriptor instanceof NodePropertyDescriptor && ((NodePropertyDescriptor) descriptor).isComposite() )
```

Comment:
Yes. We have ideas for that. Interestingly the current code represents the state after already refactoring away quite a few of these kinds of checks. We know have (I believe) few checks than stock 3.2. But this process can be taken further. We have some simple ideas that will improve this directly, and are prototyping some more fundamental changes to the descriptor approach that will reduce these checks a lot. I'd rather that work be done in later pull requests.